src/components/Greetings.js

```diff
@@ -0,0 +1,28 @@
+import React, { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { fetchGreetings } from './store/greetingsSlice';
+
+function Greetings() {
+  const dispatch = useDispatch();
+  const greetingsCont = useSelector((state) => state.greetings.greetings);
+
+  useEffect(() => {
+    dispatch(fetchGreetings());
+  }, [dispatch]);
+
+  const getRandomGreeting = () => {
+    if (greetingsCont.length > 0) {
+      const randomIndex = Math.floor(Math.random() * greetingsCont.length);
+      return greetingsCont[randomIndex].greeting;
+    }
+    return '';
+  };
+
+  return (
+    <div>
+      <h2>{getRandomGreeting()}</h2>
+    </div>
+  );
+}
+
+export default Greetings;
```

Comment:
- [ ] This code isn't working. Take in mind that you are already sending **ONE** greeting from your backend, not an array:

![image](https://github.com/abdozayan12/react-frontend/assets/77069011/a4241ec3-81a4-44c0-b8f7-93ca1be5b80b)

Then, you're setting this `Greeting` object as your state in `src/components/store/greetingsSlice.js`:

```js
{
// ...
  reducers: {
    setGreetings: (state, action) => {
      state.greetings = action.payload;
    },
// ...
}
```

And finally, in your component, you're iterating through this piece of data as if it were an array. Which ends up not showing anything.

Please consider updating the code to correctly display the greeting sent by the server! 🤓 🔧 

```suggestion
import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchGreetings } from './store/greetingsSlice';

function Greetings() {
  const dispatch = useDispatch();
  const { greeting } = useSelector((state) => state.greetings.greetings);

  useEffect(() => {
    dispatch(fetchGreetings());
  }, [dispatch]);









  return (
    <div>
      <h2>{greeting}</h2>
    </div>
  );
}

export default Greetings;
```